src/main/python/pybuilder/utils.py

```diff
@@ -194,6 +194,14 @@ def execute_command_and_capture_output(*command_and_arguments):
     return process_return_code, stdout, stderr
 
 
+def execute_commandstr_and_capture_output(command_and_arguments, env=None, cwd=None):
```

Comment:
can you de-duplicate this? You can expand the interface of `execute_command_and_capture_output` to include `env` and `cwd` without doing harm to any potential callers. Then, use `shlex` in `execute_commandstr_and_capture_output`  to take apart the `command_and_arguments` string and finally just hand over everything to `execute_command_and_capture_output`.

Reply:
Hello sorry for the late respose! Yeah if you look at my first commit this line wasn't duplicated but some older version of python doesn't allow this signature:
`def foo(*bar,other=None)`

So yeah to have the green check in this pull request I was forced to workaround and duplicate the code...